viewing_party/party.py

```diff
@@ -1,23 +1,150 @@
 # ------------- WAVE 1 --------------------
 
+from enum import unique
+
+
 def create_movie(title, genre, rating):
-    pass
+    my_dict = {}
 
-# -----------------------------------------
+    if not bool(title) or not bool(genre) or not bool(rating):
+        return None
+    else:
+        my_dict["title"] = title
+        my_dict["genre"] = genre
+        my_dict["rating"] = rating
+        
+    return my_dict
+
+def add_to_watched(user_data, movie):
+    watched_movies_list = user_data.get('watched')
+    watched_movies_list.append(movie)
+    user_data["watched"] = watched_movies_list
+    
+    return user_data
+    
+def add_to_watchlist(user_data, movie):
+    watchlist_movies_list = user_data.get('watchlist')
+    watchlist_movies_list.append(movie)
+    user_data["watchlist"] = watchlist_movies_list
+    return user_data
+    
+def watch_movie(user_data, title):
+    watchlist_movies = user_data.get('watchlist')
+    for movie in watchlist_movies:
+        if movie["title"] == title:
+            watchlist_movies.remove(movie)
+            updated_user_data = add_to_watched(user_data, movie)
+            return updated_user_data
+    return user_data
+
+
+#  --------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    watched_list = user_data.get("watched")
+    if not bool(watched_list):
+        return 0.0
+    rating_sum = 0.0
+    for movie in watched_list:
+        rating_sum += movie["rating"]
+        
+    average_rating = rating_sum/len(watched_list)
+    return average_rating
 
 
+def get_most_watched_genre(user_data):
+    watched_list = user_data.get("watched")
+    if not bool(watched_list):
+        return None
+    genre_dict = {}
+    for movie in watched_list:
+        genre = movie.get("genre")
+        if genre in genre_dict:
+            count = genre_dict[genre]
+            count += 1
+            genre_dict[genre] = count

```

Comment:
this is a little wordy! do we need to make a new variable to add `1` to itself?  let's do this:

```suggestion
            genre_dict[genre] += 1
```